sale_delivery_state/models/sale_order.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2018 Akretion (http://www.akretion.com).
+# @author Pierrick BRUN <pierrick.brun@akretion.com>
+# Copyright 2018 Camptocamp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.tools import float_compare, float_is_zero
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    delivery_state = fields.Selection(
+        [
+            ("no", "No delivery"),
+            ("unprocessed", "Unprocessed"),
+            ("partially", "Partially processed"),
+            ("done", "Done"),
+        ],
+        # Compute method have a different name then the field because
+        # the method _compute_delivery_state already exist to compute
+        # the field delivery_set
+        compute="_compute_sale_delivery_state",
+        store=True,
+    )
+
+    force_delivery_state = fields.Boolean(
```

Comment:
@rousseldenis force_delivery_state is used by Backend user to force the delivery_state to "Done" by button. And, if the user wants to revert the change, then he uses the unforce button such that it will have the info based on the Calculated field.

If we remove this field(force_delivery_state), the computation might be re-triggered for the delivery_state field based on list of dependent.

Reply:
@atchuthan Nope.

Making the delivery_state field readonly False, will allow user to change manually the delivery state. The compute won't trigger after that only if one field in the depends is changed.

But maybe this is something unwanted as we really want to force it (even if qty_delivered on lines change)

Reply:
@rousseldenis We removed force_delivery_state field as suggested.  Please review.

Reply:
@rousseldenis you think it is a good idea to remove the logic associated with force_delivery_state? 

I think that if I have decided to set the status of a sale to delivered as done manually, this status should not be recalculated if the quantity of any line is changed. 

But if this change is correct, the action_unforce_delivery_state method is not needed, as the delivery_state field is now selectable.